### PR TITLE
Splitting unary functions (in Expr) up a bit

### PR DESCRIPTION
**Pull request by balacij**
Naming may not be the best at the moment, but I'm trying to split up the unary functions up a bit (it's easier than the binary functions for now I think since we have less "groups/categories" of unary functions than binary functions).

### CODE REVIEW

File: code/drasil-lang/Language/Drasil/Expr.hs [hunk 1]
```diff
@@ -26,16 +26,23 @@ data BinOp = Frac | Pow | Subt | Eq | NEq | Lt | Gt | LEq | GEq | Impl | Iff | I
...
+  deriving Eq
 
-data BoolOper = And | Or deriving (Eq)
+data BoolOper = And | Or
+  deriving Eq
 
 -- | Unary functions
-data UFunc = Norm | Abs | Log | Ln | Sin | Cos | Tan | Sec | Csc | Cot | Arcsin
-  | Arccos | Arctan | Exp | Sqrt | Not | Neg | Dim
+data UFunc = Abs | Log | Ln | Sin | Cos | Tan | Sec | Csc | Cot | Arcsin
```
Comment (JacquesCarette) [inline code/drasil-lang/Language/Drasil/Expr.hs:36]: Just to be clear: this kind of tiny step forward is exactly the kind of thing I was thinking of.

File: code/drasil-lang/Language/Drasil/Expr/Math.hs [hunk 2]
```diff
@@ -64,7 +64,11 @@ exp = UnaryOp Exp
 
 -- | Smart constructor for the dimension of a vector
 dim :: Expr -> Expr
-dim = UnaryOp Dim
+dim = UnaryOpVec Dim
+
+-- | Smart constructor the the normal form of a vector
```
Comment (JacquesCarette) [inline code/drasil-lang/Language/Drasil/Expr/Math.hs:69]: 'the the'?

File: code/drasil-printers/Language/Drasil/Printing/Import.hs [hunk 3]
```diff
@@ -136,11 +136,11 @@ expr (UnaryOp Cot u)    sm = mkCall sm P.Cot u
 expr (UnaryOp Arcsin u) sm = mkCall sm P.Arcsin u
 expr (UnaryOp Arccos u) sm = mkCall sm P.Arccos u
 expr (UnaryOp Arctan u) sm = mkCall sm P.Arctan u
-expr (UnaryOp Dim u)    sm = mkCall sm P.Dim u
-expr (UnaryOp Not u)    sm = P.Row [P.MO P.Not, expr u sm]
 expr (UnaryOp Exp u)    sm = P.Row [P.MO P.Exp, P.Sup $ expr u sm]
 expr (UnaryOp Abs u)    sm = P.Fenced P.Abs P.Abs $ expr u sm
-expr (UnaryOp Norm u)   sm = P.Fenced P.Norm P.Norm $ expr u sm
+expr (UnaryOpB Not u)    sm = P.Row [P.MO P.Not, expr u sm]
```
Comment (JacquesCarette) [inline code/drasil-printers/Language/Drasil/Printing/Import.hs:141]: When you edit a function, it is worth trying to keep the vertical alignment. It makes reading the code later much easier.

**Review by JacquesCarette — CHANGES_REQUESTED** [review]

File: code/drasil-printers/Language/Drasil/Printing/Import.hs (hunk 3 above)
Comment (balacij) [reply code/drasil-printers/Language/Drasil/Printing/Import.hs]: I'll make sure to do this better in the future, thank you!

File: code/drasil-lang/Language/Drasil/Expr/Math.hs (hunk 2 above)
Comment (balacij) [reply code/drasil-lang/Language/Drasil/Expr/Math.hs]: Fixed it, thank you!

File: code/drasil-example/Drasil/SSP/DataDefs.hs [hunk 4]
```diff
@@ -192,7 +192,7 @@ ratioVarEqn = completeCase [case1, case2]
   where case1 = (1, sy constF)
 
         case2 = (sin (sy QM.pi_ * ((inxi slipDist - idx (sy slipDist) 0) /
-                (indxn slipDist - idx (sy slipDist) 0))), UnaryOp Not (sy constF))
+                (indxn slipDist - idx (sy slipDist) 0))), UnaryOpB Not (sy constF))
```
[the remark this answers is not in the document]
Comment (balacij) [reply code/drasil-example/Drasil/SSP/DataDefs.hs]: Thank you, I just added `not` smart constructor.

File: code/drasil-lang/Language/Drasil/Expr.hs (hunk 1 above)
Comment (balacij) [reply code/drasil-lang/Language/Drasil/Expr.hs:36]: I made a bit of a mess with the binary functions that I originally attempted, but I think I've got a better idea of how small my steps should be now. Thank you!

File: code/drasil-lang/Language/Drasil.hs [hunk 5]
```diff
@@ -1,7 +1,7 @@
 {- re-export many things to simplify external use -}
 module Language.Drasil (
   -- Expr
-  Expr(..), BinOp(..), UFunc(..), ArithOper(..), BoolOper(..), DerivType(..)
+  Expr(..), BinOp(..), UFunc(..), UFuncB(..), UFuncVec(..), ArithOper(..), BoolOper(..), DerivType(..)
```
[the remark this answers is not in the document]
Comment (balacij) [reply code/drasil-lang/Language/Drasil.hs:4]: Ah, that's interesting. Should I just re-export it naively through `drasil-lang/Language/Drasil/Development.hs` then?

If so, should we be removing `UFunc` and `BinOp` as well?

[the remark this answers is not in the document]
Comment (JacquesCarette) [reply code/drasil-lang/Language/Drasil.hs:4]: Yes, re-export it naively through there. Over time, I think we'll be able to provide enough traversal combinators to even remove that, but not right away.

In time, yes. We need to have the examples not use concrete data constructors at all. Incrementally.

File: code/drasil-example/Drasil/SSP/DataDefs.hs (hunk 4 above)
[the remark this answers is not in the document]
Comment (balacij) [reply code/drasil-example/Drasil/SSP/DataDefs.hs]: (changed to `not_`)

File: code/drasil-lang/Language/Drasil.hs (hunk 5 above)
[the remark this answers is not in the document]
Comment (balacij) [reply code/drasil-lang/Language/Drasil.hs]: Sounds good, I just re-exported it in c7326fa40701a7e3ce4ad71d7ce8131c6a47d051

**Review by JacquesCarette — APPROVED** [review]

**Pull request merged**